### PR TITLE
Add support for surrogate pairs

### PR DESCRIPTION
**Pull request by kevva**

### CODE REVIEW

**Comment by Qix-** [comment]
lgtm if it passes :)

File: index.js [hunk 1]
```diff
@@ -38,15 +38,20 @@ const ESCAPE_CODES = new Map([
 const wrapAnsi = code => `${ESCAPES[0]}[${code}m`;
 
 module.exports = (str, begin, end) => {
-	end = end || str.length;
+	const arr = Array.from(str.normalize());
+
+	end = end || arr.length;
```
Comment (Qix-) [inline index.js]: Will `end` ever be `0` here?

Comment (kevva) [inline index.js]: I guess, if the user sets it to `0` or the string is empty. Why?

Comment (kevva) [inline index.js]: Oh, you meant in regards to `||` (need sleep, lol). Yeah, we might have to fix that.

Comment (Qix-) [inline index.js]: Yeah, might be safe to do `end === undefined ? arr.length : end`.

Comment (kevva) [inline index.js]: Maybe `typeof end === 'number' ? end : arr.length`? Then we make sure `end` won't fail because of incorrect input.

**Comment by sindresorhus** [comment]
Thanks for fixing Kevin :)

**Pull request merged**